File: src/main/java/me/itzisonn_/meazy/registry/CommandRegistry.java
package me.itzisonn_.meazy.registry;

import me.itzisonn_.meazy.command.AbstractCommand;
import me.itzisonn_.registry.RegistryEntry;
import me.itzisonn_.registry.RegistryIdentifier;
import me.itzisonn_.registry.multiple_entry.SetRegistry;

public class CommandRegistry extends SetRegistry<AbstractCommand> {
    @Override
    public void register(RegistryIdentifier identifier, AbstractCommand value, boolean overridable) {
        for (RegistryEntry<AbstractCommand> entry : getEntries()) {
            if (entry.getValue().getName().equals(value.getName()) && !entry.isOverrideable()) {
                throw new IllegalArgumentException("AbstractCommand with name " + value.getName() + " has already been registered");
            }
        }

        super.register(identifier, value, overridable);
    }
}

File: src/main/java/me/itzisonn_/meazy/registry/CommandRegistry.kt
package me.itzisonn_.meazy.registry

import me.itzisonn_.meazy.command.Command
import me.itzisonn_.registry.RegistryIdentifier
import me.itzisonn_.registry.multiple_entry.SetRegistry

class CommandRegistry : SetRegistry<Command>() {
    override fun register(identifier: RegistryIdentifier, value: Command, overridable: Boolean) {
        for (entry in entries) {
            require(!(entry.getValue().id == value.id && !entry.isOverrideable)) {
                "Command with id '${value.id}' has already been registered"
            }
        }

        super.register(identifier, value, overridable)
    }
}
